packages/@ember/-internals/extension-support/lib/data_adapter.js

```diff
@@ -53,6 +163,9 @@ export default EmberObject.extend({
   init() {
     this._super(...arguments);
     this.releaseMethods = emberA();
+    this.recordsWatchers = new Map();
+    this.typeWatchers = new Map();
+    this.flushWatchers = this.flushWatchers.bind(this);
```

Comment:
Why bind like this?

Reply:
Well, you can't do `this.flushWatchers = () => this.flushWatchers();`, that'll cause infinite recursion. Open to suggestions

Reply:
When I commented I hadn't noticed the callback usage. I think I've usually seen `this._boundFlushWatchers = () => this.flushWatchers();` (or use `.bind`). 

I didn't don't care about this though, so if you don't think it should change feel free to resolve.